src/transaction.js

```diff
@@ -57,20 +57,20 @@ Transaction.fromBuffer = function (buffer, __disableAssert) {
 
   var vinLen = readVarInt()
   for (var i = 0; i < vinLen; ++i) {
-    var hash = readSlice(32)
+    var txHash = readSlice(32)
 
-    if (Transaction.isCoinbaseHash(hash)) {
+    if (Transaction.isCoinbaseHash(txHash)) {
       tx.ins.push({
-        hash: hash,
-        index: readUInt32(),
-        script: readGenerationScript(),
+        txHash: txHash,
```

Comment:
In regards to https://github.com/bitcoinjs/bitcoinjs-lib/pull/421 and storing `txId` here as well as `txHash`, as it used quite often... perhaps we could add `txId` as a computed property?
I'd just like to discourage it being computed constantly,  and computed properties can be confusing with a type system.